lib/ecto/adapters/myxql/connection.ex

```diff
@@ -222,10 +222,16 @@ if Code.ensure_loaded?(MyXQL) do
     end
 
     @impl true
-    # DB explain opts are deprecated, so they aren't used to build the explain query.
+    # DB explain opts, except format, are deprecated.
     # See Notes at https://dev.mysql.com/doc/refman/5.7/en/explain.html
     def explain_query(conn, query, params, opts) do
-      case query(conn, build_explain_query(query), params, opts) do
+      {explain_opts, opts} = Keyword.split(opts, ~w[format]a)
+      json_format? = {:format, :json} in explain_opts
+
+      case query(conn, build_explain_query(query, explain_opts), params, opts) do
+        {:ok, %MyXQL.Result{rows: rows}} when json_format? ->
+          {:ok, List.flatten(rows)}
```

Comment:
I think we can decode this if we know it will be valid json returned? The driver can't because it can't tell we are doing a json explain query but here we know.

Reply:
We can. `Jason` is just a test+doc dependency though. Do you think it's worth changing that?

Reply:
We don't have to make `Jason` required. We can use the user-specified JSON library and fall back to Jason, like here: https://github.com/elixir-ecto/ecto_sql/blob/master/lib/ecto/adapters/myxql/connection.ex#L947.

Reply:
I should probably get a second opinion since this situation is a bit strange.

@josevalim For MySQL when you specify the JSON explain format, it is returned as text type so MyXQL can't convert it into a map. Do you think it's ok to try to decode it here with the JSON library specified by the user? My thinking is it's probably better UX to not have to not have to decode it and also make the return the same as Postgres.

Reply:
It is probably fine to decide if the format is json and we get a string back, yeah.

Reply:
Great! Agree that it's better UX